Java/PoolAndResourceFile/src/main/java/PoolAndResourceFile.java

```diff
@@ -0,0 +1,299 @@
+
+        return client.getPoolOperations().getPool(poolId);
+    }
+
+    // Create blob container in order to upload file
+    private static CloudBlobContainer createBlobContainer(String storageAccountName, String storageAccountKey) throws URISyntaxException, StorageException {
+        String CONTAINER_NAME = "poolsandresourcefiles";
+
+        // Create storage credential from name and key
+        StorageCredentials credentials = new StorageCredentialsAccountAndKey(storageAccountName, storageAccountKey);
+
+        // Create storage account
+        CloudStorageAccount storageAccount = new CloudStorageAccount(credentials);
+
+        // Create the blob client
+        CloudBlobClient blobClient =  storageAccount.createCloudBlobClient();
+
+        // Get a reference to a container.
+        // The container name must be lower case
+        return blobClient.getContainerReference(CONTAINER_NAME);
+    }
+
+    // Upload file to blob container and return sas key
+    private static String uploadFileToCloud(CloudBlobContainer container) throws URISyntaxException, IOException, InvalidKeyException, StorageException {
+        String FILE_NAME = "test.txt";
+        String FILE_PATH = "./" + FILE_NAME;
+
+        // Create the container if it does not exist.
+        container.createIfNotExists();
+
+        // Upload file
+        CloudBlockBlob blob = container.getBlockBlobReference(FILE_NAME);
+        File source = new File(FILE_PATH);
+        blob.upload(new FileInputStream(source), source.length());
+
+        // Create policy with 1 day read permission
+        SharedAccessBlobPolicy policy = new SharedAccessBlobPolicy();
+        EnumSet<SharedAccessBlobPermissions> perEnumSet = EnumSet.of(SharedAccessBlobPermissions.READ);
+        policy.setPermissions(perEnumSet);
+
+        Calendar c = Calendar.getInstance();
+        c.setTime(new Date());
+        c.add(Calendar.DATE, 1);
+        policy.setSharedAccessExpiryTime(c.getTime());
+
+        // Create SAS key
+        String sas = blob.generateSharedAccessSignature(policy, null);
+        return blob.getUri() + "?" + sas;
+    }
+
+    // Create a job with a single task
+    private static void submitJobAndAddTask(BatchClient client, CloudBlobContainer container, String poolId, String jobId) throws BatchErrorException, IOException, StorageException, InvalidKeyException, URISyntaxException {
+        String FILE_NAME = "mytest.txt";
+
+        // Create job run at the specified pool
+        PoolInformation poolInfo = new PoolInformation();
+        poolInfo.setPoolId(poolId);
+        client.getJobOperations().createJob(jobId, poolInfo);
+
+        // Create task
+        TaskAddParameter taskToAdd = new TaskAddParameter();
+        taskToAdd.setId("mytask");
+        taskToAdd.setCommandLine(String.format("cat %s", FILE_NAME));
+
+        String sas = uploadFileToCloud(container);
+
+        // Associate resource file with task
+        ResourceFile file = new ResourceFile();
```

Comment:
minor: Does this by chance have a constructor generated which just takes the file path and blob source?